### PR TITLE
[Agrifood] Adding client id and client secret parameter in scenes create job req…

### LINKED ISSUE #23007: [Agrifood ] API Review - Adding client id and client secret parameter
New API Review meeting has been requested.

**Service Name**: Microsoft Azure Data Manager for Agriculture
**Review Created By**: Bhanu Prakash Chinthalapally
**Review Date**: 3/16/2023 8:00 AM PT
**PR**: https://github.com/Azure/azure-rest-api-specs/pull/23024
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: We are changing existing Scenes APIs to support BYOL where take extra input params of client key/secrets in the existing PUT /scenes/satellite/ingest-data/{jobId} API.

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=8d555922-1abf-ed11-83ff-000d3a3418aa)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by bhargav-kansagara**
# Data Plane API - Pull Request
- Adding provider client secret and client id to support BYOL for ingesting satellite data

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json [hunk 1]
```diff
@@ -23341,7 +23349,7 @@
           "description": "Provider of satellite data. Available Value: Microsoft.",
```
Comment (mikekistler) [inline specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json:23349]: You probably need to update this comment.

**Comment by weidongxu-microsoft** [comment]
I thought 2022-11-01-preview already under public preview? Adding property would then require a new api-version?

And one change seems breaks?

**Comment by bhargav-kansagara** [comment]
@weidongxu-microsoft Yes, it is a public preview version but we have got this guidelines from CELA team and we won't be supporting the existing API from next week onwards. I can add a new API version but this API will still be blocked in existing API versions.

**Comment by weidongxu-microsoft** [comment]
OK, I will leave this to Mike, as anyway this triggers BreakingChange CI and I won't be able to do much.

File: specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json [hunk 2]
```diff
@@ -21973,6 +21973,14 @@
         "data": {
           "$ref": "#/definitions/SatelliteData"
         },
+        "providerClientId": {
```
Comment (BlackRider97) [inline specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json:21976]: order of parameters should be lexicographically.

File: specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json [hunk 3]
```diff
@@ -21973,6 +21973,14 @@
...
+          "description": "Provider client id for authenticating to provider to pull satellite data.",
+          "type": "string"
+        },
+        "providerClientSecret": {
```
Comment (BlackRider97) [inline specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json:21980]: should be marked as required ?

File: specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json (hunk 2 above)
Comment (bhargav-kansagara) [reply specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json:22203]: That will lead to a lot of changes, not doing in this PR.

File: specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json (hunk 3 above)
Comment (bhargav-kansagara) [reply specification/agrifood/data-plane/Microsoft.AgFoodPlatform/preview/2022-11-01-preview/agfood.json:22207]: Done

**Review by markweitzel — APPROVED** [review]
## API Stewardship Board Review 3-16-23
Per: #23007 approved for public preview.

**Review by heaths — APPROVED** [review]

**Pull request closed without merging**